### PR TITLE
Possibility to enable/disable Indexed database in runtime

### PR DESCRIPTION
**Pull request by vertexodessa**

### CODE REVIEW

File: Source/WebCore/page/SecurityOrigin.h [hunk 1]
```diff
@@ -141,14 +141,17 @@ class SecurityOrigin : public ThreadSafeRefCounted<SecurityOrigin> {
 
     void setStorageBlockingPolicy(StorageBlockingPolicy policy) { m_storageBlockingPolicy = policy; }
 
+    void setShouldBlockDatabaseStorage(bool block) { m_shouldBlockDatabaseStorage = block; }
+    bool shouldBlockDatabaseStorage() { return m_shouldBlockDatabaseStorage; }
```
Comment (calvaris) [inline Source/WebCore/page/SecurityOrigin.h]: make the method const, as the other ones returning bool, please

Comment (vertexodessa) [reply Source/WebCore/page/SecurityOrigin.h]: thanks, fixed

File: Source/WebKit2/Shared/WebPreferencesDefinitions.h [hunk 2]
```diff
@@ -110,7 +110,7 @@
     macro(JavaEnabledForLocalFiles, javaEnabledForLocalFiles, Bool, bool, false, "", "") \
     macro(OfflineWebApplicationCacheEnabled, offlineWebApplicationCacheEnabled, Bool, bool, false, "", "") \
     macro(LocalStorageEnabled, localStorageEnabled, Bool, bool, true, "", "") \
-    macro(DatabasesEnabled, databasesEnabled, Bool, bool, true, "", "") \
```
Comment (zdobersek) [inline Source/WebKit2/Shared/WebPreferencesDefinitions.h]: This should default to true, and WKPreferencesSetDatabasesEnabled() should be called instead to disable the preference.

File: Source/WebKit2/WebProcess/WebPage/wpe/WebPageWPE.cpp [hunk 3]
```diff
@@ -54,6 +54,7 @@ void WebPage::platformPreferencesDidChange(const WebPreferencesStore& store)
     m_page->settings().setAllowRunningOfInsecureContent(store.getBoolValueForKey(WebPreferencesKey::allowRunningOfInsecureContentKey()));
     m_page->settings().setAllowDisplayOfInsecureContent(store.getBoolValueForKey(WebPreferencesKey::allowDisplayOfInsecureContentKey()));
     m_page->settings().setScrollToFocusedElementEnabled(store.getBoolValueForKey(WebPreferencesKey::scrollToFocusedElementEnabledKey()));
+    m_page->settings().setShouldBlockDatabaseStorage(!store.getBoolValueForKey(WebPreferencesKey::databasesEnabledKey()));
```
Comment (zdobersek) [inline Source/WebKit2/WebProcess/WebPage/wpe/WebPageWPE.cpp]: All the changes to Document, SecurityOrigin and Settings aren't needed -- I think it should be simpler to add a new RuntimeEnabledFeature entry that in WebPage::updatePreferences() is set to the negative value of WebPreferencesKey::databasesEnabledKey(), and that is then checked in shouldThrowSecurityException() in IDBFactory.cpp where it causes the exception throw if it's set to true.

Note that DatabaseManager::singleton().setIsAvailable() is already called in WebPage::updatePreferences(), where it's set to the value of databasesEnabledKey(). This results in a security exception being thrown in DOMWindowWebDatabase::openDatabase() just the same as if the SecurityOrigin::canAccessDatabase() call returned false there.

Comment (vertexodessa) [reply Source/WebKit2/WebProcess/WebPage/wpe/WebPageWPE.cpp]: @zdobersek 
Hi Zan,
don't we need to support separate settings for multiple web pages running in the same web process?
(see WebProcessPool::createNewWebProcessRespectingProcessCountLimit https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/master/Source/WebKit2/UIProcess/WebProcessPool.cpp#L768)

Comment (vertexodessa) [reply Source/WebKit2/WebProcess/WebPage/wpe/WebPageWPE.cpp]: In any case to speed up the code review, I've re-raised pull request according to your comments. Please let me know how you think we should proceed.
Thanks!

**Pull request merged**